receiver/awscontainerinsightreceiver/internal/ecsInfo/cgroup_test.go

```diff
@@ -0,0 +1,216 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ecsinfo
+
+import (
+	"context"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+)
+
+type MockTaskInfo struct {
+	tasks            []ECSTask
+	runningTaskCount int64
+}
+
+func (ii *MockTaskInfo) getRunningTaskCount() int64 {
+	return ii.runningTaskCount
+}
+func (ii *MockTaskInfo) getRunningTasksInfo() []ECSTask {
+
+	return ii.tasks
+}
+
+type MockInstanceInfo struct {
+	clusterName string
+	instanceID  string
+}
+
+func (ii *MockInstanceInfo) GetClusterName() string {
+	return ii.clusterName
+}
+func (ii *MockInstanceInfo) GetContainerInstanceID() string {
+	return ii.instanceID
+}
+
+func TestGetCGroupPathForTask(t *testing.T) {
+	cgroupMount := "test"
+	controller := "cpu"
+	taskID := "test1"
+	clusterName := "myCluster"
+	result, _ := getCGroupPathForTask(cgroupMount, controller, taskID, clusterName)
+	assert.Equal(t, path.Join(cgroupMount, controller, "ecs", taskID), result)
+
+	taskID = "test4"
+	result, _ = getCGroupPathForTask(cgroupMount, controller, taskID, clusterName)
+	assert.Equal(t, path.Join(cgroupMount, controller, "ecs", clusterName, taskID), result)
+
+	taskID = "test5"
+	result, err := getCGroupPathForTask(cgroupMount, controller, taskID, clusterName)
+	assert.Equal(t, "", result)
+	assert.NotNil(t, err)
+}
+
+func TestGetCGroupPathFromARN(t *testing.T) {
```

Comment:
Can you rewrite this as a table based test?  That should make it easier to read and to figure out which test case is failing.  The other tests might benefit from this as well.

Reply:
Sure, will change the unit test code